ratarmount.py

```diff
@@ -403,6 +403,7 @@ def __init__(
             self._createIndex(fileObject)
             # return here because we can't find a save location without any identifying name
             return
+        self.tarFileName = tarFileName
```

Comment:
Please move it to the else branch of `if not fileObject` on line 369. As it stands, you are brute-force overwriting a possibly already valid value for `self.tarFileName`